src/site/controller.ts

```diff
@@ -3,44 +3,187 @@
  *
  * This is where we handle basic functions of sites.
  */
-
 import { Request, Response, Router } from 'express';
+import { ObjectId } from 'mongoose';
+import { UserModel } from '../user/model';
+import { checkId, ExpressHandler, HTTPError } from '../utils';
+import { SiteModel } from './model';
+
+const SiteRouter = Router();
+export default SiteRouter;
+
+SiteRouter.get('/', getAllSites);
+SiteRouter.put('/', createSite);
+SiteRouter.delete(
+  '/',
+  async (req, res, next) =>
```

Comment:
All of these middlewares could be written as `(req, res, next) => checkId(res, next, ...),` due to the fact that `next` is what will let express know to continue to the endpoint handler. There is no need to write `async (req, res, next) => await f()`, as the `async/await` will have no effect.

Reply:
Removed async/await from middleware